crates/client/src/op/update_message.rs

```diff
@@ -550,3 +550,32 @@ pub fn delete_all(name_of_records: Name, zone_origin: Name, dns_class: DNSClass)
 
     message
 }
+
+// not an update per-se, but it fits nicely with other functions here
+/// Download all records from a zone using a AXFR query.
+///
+/// # Arguments
+/// * `zone_origin` - the zone name to update, i.e. SOA name
+pub fn zone_transfert(zone_origin: Name) -> Message {
```

Comment:
again, reason for the `t` on the end?

Also, do you think this can (in a later revision) be expanded to include `IXFR`, I just want to make sure the APIs would all align... I don't see any reason why not. We could just have a `zone_transfer_since` and that would instead send an `IXFR` instead.

Reply:
Up to and including a4d59ea, it had support for IXFR by taking an `Option<SOA>` , but considering how IXFR work, I'm not sure it's the best interface. Assuming client last known serial is 2, and server serial is 4, a sample answer might look like this:
```sh
- SOA[serial=5]
- SOA[serial=2]
- A[example.com=1.2.3.4] # deleted
- SOA[serial=3]
- A[example.com=10.11.12.13] # added
- SOA[serial=3]
- A[example.com=10.11.12.13] # deleted
- SOA[serial=4]
- SOA[serial=4]
```
Where the A record is modified when incrementing from 2 to 3, and then deleted when incrementing from 3 to 4.
In the end, for the client, the import part is probably that `A[example.com=1.2.3.4]` was deleted, and knowing that another record existed for a brief instant is generally not useful. I think having a function returning 2 sets of records, one set of deleted and one of added, would probably be easier to use. For `update_message::zone_transfer` it changes nothing (it's just returning the message to send to the server), but for `Client::zone_transfer`, IXFR can either be compatible with what is there for AXFR, and return the raw answer, or be not compatible, and parse the result in something easier to work with

Reply:
I would say, let's keep it as simple as possible.

This looks like good analysis though. Thank you.

Reply:
So keep the same interface for IXFR and AXFR, and maybe add somewhere an helper function returning the added and removed sets given a `Vec<DnsResponse>`?

Reply:
Right, I think that's correct. We'd probably need to wrap the responses in something to demarcate the deletes, vs. additions. Basically, I think having a stream of updates is fine, and then allow the consumer to determine what to do with the records.